### PR TITLE
Kameraga boglab

### PR DESCRIPTION
**Pull request by orifxolbekov**
Kamera orqali gradioda ishladik

### CODE REVIEW

**Comment by orifxolbekov** [comment]
```
Traceback (most recent call last):
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/gradio/routes.py", line 337, in run_predict
    output = await app.get_blocks().process_api(
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/gradio/blocks.py", line 1015, in process_api
    result = await self.call_function(
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/gradio/blocks.py", line 833, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/ubuntuuser/proyektlar/python/projects/gradio_apps/kameraga_boglab.py", line 38, in flip
    rasm1_embedding, rasm1_koordinatalar = embedding_olish(rasm1, yuz_aniqlagich_modul, embedding_modul)
  File "/home/ubuntuuser/proyektlar/python/projects/gradio_apps/kameraga_boglab.py", line 23, in embedding_olish
    koordinatalar = aniqlash(model_aniq, rasm)[0]
  File "/home/ubuntuuser/proyektlar/python/projects/gradio_apps/refaktor/yuzbib.py", line 107, in aniqlash
    kordinatalar, landmarklar = model.detect(rasm)
  File "/home/ubuntuuser/miniconda3/envs/yuz_taqqoslash/lib/python3.9/site-packages/insightface/model_zoo/retinaface.py", line 211, in detect
    im_ratio = float(img.shape[0]) / img.shape[1]
AttributeError: 'str' object has no attribute 'shape'

```
xatolik berdi

**Comment by orifxolbekov** [comment]
tekshirib kordim ishladi

**Review by NuriddinAI — COMMENTED** [review]
Traceback (most recent call last):
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/gradio/routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/gradio/blocks.py", line 1015, in process_api
    result = await self.call_function(fn_index, inputs, iterator, request)
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/gradio/blocks.py", line 856, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/nuriddin/ochiqai/python/projects/gradio_apps/kameraga_boglab.py", line 49, in kamera
    baza_rasm_kordinata = aniqlash(aniqlagich_model, baza_rasm)
  File "/home/nuriddin/ochiqai/python/projects/gradio_apps/refaktor/yuzbib.py", line 107, in aniqlash
    kordinatalar, landmarklar = model.detect(rasm)
  File "/home/nuriddin/miniconda3/envs/ochiqai2/lib/python3.8/site-packages/insightface/model_zoo/retinaface.py", line 211, in detect
    im_ratio = float(img.shape[0]) / img.shape[1]
AttributeError: 'NoneType' object has no attribute 'shape'

**Comment by NuriddinAI** [comment]
Yuqoridagi hatolikni oldim

**Comment by orifxolbekov** [comment]
bazaga ozini yukladingmi

**Comment by NuriddinAI** [comment]
bazada mavjud emas deb qaytaradiku, programmani o'zini ishlatolmadim

**Comment by orifxolbekov** [comment]
@NuriddinAI ishladimi

File: projects/gradio_apps/kameraga_boglab.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import cv2
+import gradio as gr
+import os
+from projects.gradio_apps.refaktor.yuzbib import (
+    model_yuklash, aniqlash, vector_taq, chizish
+)
+#----------------------- Proyekt: ------------------------------------
+# Kamera orqali rasm kiritiladi, uni bazadagi rasmlar bilan solishtiradi.
+#   Agar: kiritilgan rasm bazada mavjud bo'lsa
+#       uni kimligini qaytaradi
```
Comment (ochiqai) [inline projects/gradio_apps/kameraga_boglab.py:10]:
```suggestion
#       uni kimligini rasm balantida chap tarafda ko'rsatadi
```

File: projects/gradio_apps/kameraga_boglab.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+def kamera(rasm):
+    """
+    Kameradan rasmni olib bazadagi rasm bilan solishtiradi
+    Parameters
+    ----------
+    rasm:'numpy.ndarray'
+
+    Returns
+    -------
+    natija qaytariladi
```
Comment (ochiqai) [inline projects/gradio_apps/kameraga_boglab.py:41]:
```suggestion
    natija : 'numpy.ndarray'
     yuz kimga tegishligi yozilgan rasm
```

File: projects/gradio_apps/kameraga_boglab.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+baza_rasm = cv2.imread("./refaktor/rasmlar/Orif.png")
+
+def kamera(rasm):
+    """
+    Kameradan rasmni olib bazadagi rasm bilan solishtiradi
+    Parameters
+    ----------
+    rasm:'numpy.ndarray'
```
Comment (ochiqai) [inline projects/gradio_apps/kameraga_boglab.py:37]:
```suggestion
    rasm : 'numpy.ndarray'
       [b, e, c] o'lchovli matritsa. masalan [200, 400, 3]
```

File: projects/gradio_apps/kameraga_boglab.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+fontScale = 0.8
+color = (255, 0, 0)
+thickness = 1
+
+
+baza_rasm = cv2.imread("./refaktor/rasmlar/Orif.png")
+
+def kamera(rasm):
+    """
+    Kameradan rasmni olib bazadagi rasm bilan solishtiradi
```
Comment (ochiqai) [inline projects/gradio_apps/kameraga_boglab.py:34]:
```suggestion
    Kameradan rasmni olib bazadagi rasm bilan solishtiradi va ramsdagu yuz kimga tegishli
    ekanligini rasm tepasiga chap tarafga yozadi.
```

File: projects/gradio_apps/kameraga_boglab.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    ----------
+    rasm:'numpy.ndarray'
+
+    Returns
+    -------
+    natija qaytariladi
+    """
+
+    aniqlagich_model = model_yuklash(turi="aniqlagich")
+    embedding_modul = model_yuklash(turi="embedding")
```
Comment (ochiqai) [inline projects/gradio_apps/kameraga_boglab.py:45]:
```suggestion
    embedding_model = model_yuklash(turi="embedding")
```

**Review by ochiqai — CHANGES_REQUESTED** [review]

**Review by ochiqai — COMMENTED** [review]
Link qo'yishimiz kerak - README.md

**Review by ochiqai — APPROVED** [review]
yaxshi

**Pull request merged**